libs/MeshKernelApi/src/MeshKernel.cpp

```diff
@@ -1710,6 +1711,106 @@ namespace meshkernelapi
         return exitCode;
     }
 
+    MKERNEL_API int mkernel_splines_snap_to_landboundary(int meshKernelId,
+                                                         const GeometryList& land,
+                                                         GeometryList& splines,
+                                                         int startSplineIndex,
+                                                         int endSplineIndex)
+    {
+        int exitCode = Success;
+
+        try
+        {
+            if (meshKernelState.count(meshKernelId) == 0)
+            {
+                throw std::invalid_argument("MeshKernel: The selected mesh kernel id does not exist.");
+            }
+
+            if (startSplineIndex > endSplineIndex)
+            {
+                throw std::invalid_argument("Invalid spline range: " + std::to_string(startSplineIndex) + " > " + std::to_string(endSplineIndex));
+            }
+
+            if (land.num_coordinates == 0)
+            {
+                throw std::invalid_argument("Land boundary has no point values.");
+            }
+
+            if (land.coordinates_x == nullptr || land.coordinates_y == nullptr)
+            {
+                throw std::invalid_argument("Land boundary data is null.");
+            }
+
+            if (splines.num_coordinates == 0)
+            {
+                throw std::invalid_argument("Spline has no point values.");
+            }
+
+            if (splines.coordinates_x == nullptr || splines.coordinates_y == nullptr)
+            {
+                throw std::invalid_argument("Spline data is null.");
+            }
+
+            std::vector<meshkernel::Point> landBoundaryPoints(land.num_coordinates);
+            std::vector<meshkernel::Point> splinePoints(splines.num_coordinates);
+
+            //--------------------------------
+            // Copy points from parameters to create land boundary and splines
+
+            for (int i = 0; i < land.num_coordinates; ++i)
+            {
+                landBoundaryPoints[i] = meshkernel::Point({land.coordinates_x[i], land.coordinates_y[i]});
+            }
+
+            for (int i = 0; i < splines.num_coordinates; ++i)
```

Comment:
what if multiple splines are provided? start/end index refer to the selected spline corner's point